airflow/serialization/serialized_objects.py

```diff
@@ -1462,9 +1462,15 @@ def deserialize_dag(cls, encoded_dag: dict[str, Any]) -> SerializedDAG:
                 v = set(v)
             elif k == "tasks":
                 SerializedBaseOperator._load_operator_extra_links = cls._load_operator_extra_links
-
-                v = {task["task_id"]: SerializedBaseOperator.deserialize_operator(task) for task in v}
+                tasks = {}
+                for obj in v:
+                    if obj.get(Encoding.TYPE) == DAT.OP:
+                        deser = SerializedBaseOperator.deserialize_operator(obj[Encoding.VAR])
+                        tasks[deser.task_id] = deser
+                    else:  # this is backcompat for pre-2.10
+                        tasks[obj["task_id"]] = SerializedBaseOperator.deserialize_operator(obj)
```

Comment:
@potiuk @uranusjr  this is necessatated by the change to run ser op through `cls._encode` (like we already do for everything else)

Reply:
note also that if we merge this PR with these changes, then we can remove this logic in other pr : https://github.com/apache/airflow/pull/38567/files#diff-807ca0a4fd53aeb41166621c9842b0f89b7931fc64e9a60befa36c776db45efaR664

Reply:
Don’t we always require (?) reserialising when Airflow is upgraded? This wouldn’t be needed if that’s the case.

Reply:
for better or worse, default setting is to _not_ reserialize...

https://github.com/apache/airflow/blob/fecc1ed8eff5192818bbe04cbbdfe9585eaab583/airflow/cli/cli_config.py#L646-L653